x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -484,24 +489,58 @@ export class SessionIndex {
       });
     }
 
+    const operations: Array<Required<Pick<BulkOperationContainer, 'delete'>>> = [];
     try {
-      const { body: response } = await this.options.elasticsearchClient.deleteByQuery(
-        {
-          index: this.indexName,
-          refresh: true,
-          body: { query: { bool: { should: deleteQueries } } },
-        },
-        { ignore: [409, 404] }
-      );
-
-      if (response.deleted! > 0) {
-        this.options.logger.debug(
-          `Cleaned up ${response.deleted} invalid or expired session values.`
+      const { body: searchResponse } =
+        await this.options.elasticsearchClient.search<SessionIndexValue>(
+          {
+            index: this.indexName,
+            body: { query: { bool: { should: deleteQueries } } },
+            _source_includes: 'usernameHash,provider',
+          },
+          { ignore: [409, 404] }
         );
-      }
+      searchResponse.hits.hits.forEach(({ _id, _source }) => {
+        const { usernameHash, provider } = _source!;
+        this.options.auditLogger.log(
+          sessionCleanupEvent({ sessionId: _id, usernameHash, provider })
+        );
+        operations.push({ delete: { _id } });
+      });
     } catch (err) {
-      this.options.logger.error(`Failed to clean up sessions: ${err.message}`);
+      this.options.logger.error(`Failed to look up invalid or expired sessions: ${err.message}`);
       throw err;
     }
+
+    if (operations.length > 0) {
+      try {
+        const { body: deleteResponse } = await this.options.elasticsearchClient.bulk({
+          index: this.indexName,
+          operations,
+        });
```

Comment:
It would probably be a good idea to specify `refresh: false` here; this is the default option, but it would signal to future maintainers that we made an [intentional decision not to refresh](https://www.elastic.co/guide/en/elasticsearch/reference/8.0/docs-refresh.html).